api/types/lock.go

```diff
@@ -45,6 +45,15 @@ type Lock interface {
 	// SetLockExpiry sets the lock's expiry.
 	SetLockExpiry(*time.Time)
 
+	// CreatedOn returns the time the lock was created.
+	CreatedOn() *time.Time
```

Comment:
This API would be easier to work with if it was just plain old `time.Time` values (see nullable comment above)